src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java

```diff
@@ -650,6 +663,30 @@ public Collection<NodeProvisioner.PlannedNode> provision(
         return Collections.emptyList();
     }
 
+    private static void checkServerCertificateFIPS(String serverCertificate) {
+        try {
+            CertificateFactory certificateFactory = CertificateFactory.getInstance("X.509");
+            X509Certificate certificate = (X509Certificate) certificateFactory.generateCertificate(new ByteArrayInputStream(serverCertificate.getBytes(UTF_8)));
+            PublicKey publicKey = certificate.getPublicKey();
+            if (publicKey instanceof RSAPublicKey) {
+                if (((RSAPublicKey)publicKey).getModulus().bitLength() < 2048) {
+                    throw new IllegalArgumentException(Messages.KubernetesCloud_serverCertificateKeySize());
+                }
+            } else if (publicKey instanceof DSAPublicKey) {
+                if (((DSAPublicKey)publicKey).getParams().getP().bitLength() < 2048) {
+                    throw new IllegalArgumentException(Messages.KubernetesCloud_serverCertificateKeySize());
+                }
+            } else if (publicKey instanceof ECPublicKey) {
+                if (((ECPublicKey)publicKey).getParams().getCurve().getField().getFieldSize() < 224) {
+                    throw new IllegalArgumentException(Messages.KubernetesCloud_serverCertificateKeySizeEC());
+                }
+            }
+
+        }catch (CertificateException ex) {
+            throw new IllegalStateException(ex);
+        }
```

Comment:
I am pretty sure you will get a runtime exception when running with BouncyCastle FIPS not a CertificateException, in which cast that will bubble up and out.